src/handle_provide_parameter.c

```diff
@@ -727,11 +731,46 @@ static void handle_execute(ethPluginProvideParameter_t *msg, context_t *context)
             }
             break;
 
+            // ###################
+            // Parsing PAY_PORTION
+            // ###################
+
+        case INPUT_PAY_PORTION_LENGTH:
+            context->next_param = INPUT_PAY_PORTION_TOKEN;
+            break;
+        case INPUT_PAY_PORTION_TOKEN:
+            context->next_param = INPUT_PAY_PORTION_RECIPIENT;
+            break;
+        case INPUT_PAY_PORTION_RECIPIENT:
+            context->next_param = INPUT_PAY_PORTION_AMOUNT;
+            break;
+        case INPUT_PAY_PORTION_AMOUNT:
+            // empty statement because switch / case is wonky
+            ;
```

Comment:
Not sure I understand the problem here? What is wonky??

Reply:
C doesn't support variable declaration as first statement of a `case`

Reply:
Why not putting the declarations at the begining of the function, or include the case inside {}?

Reply:
Done